packages/p2p/src/pages/my-ads/ad-type-section/ad-type-section.tsx

```diff
@@ -0,0 +1,308 @@
+                general_store.showModal({
+                    key: 'AdCancelModal',
+                    props: {
+                        message: 'If you choose to cancel, the edited details will be lost.',
+                        onConfirm: () => {
+                            my_ads_store.setShowEditAdForm(false);
+                        },
+                        title: 'Cancel your edits?',
+                    },
+                });
+            } else {
+                my_ads_store.setShowEditAdForm(false);
+            }
+        } else {
+            general_store.showModal({
+                key: 'AdCancelModal',
+                props: {
+                    message: "If you choose to cancel, the details you've entered will be lost.",
+                    onConfirm: () => {
+                        my_ads_store.setApiErrorMessage('');
+                        floating_rate_store.setApiErrorMessage('');
+                        my_ads_store.setShowAdForm(false);
+                        buy_sell_store.setCreateSellAdFromNoAds(false);
+                    },
+                    title: 'Cancel ad creation?',
+                },
+            });
+        }
+    };
+
+    return (
+        <>
+            {!is_edit && (
+                <Field name='type'>
+                    {({ field }) => (
+                        <RadioGroup
+                            {...field}
+                            className='ad-type-section__radio-group'
+                            name='type'
+                            onToggle={event => onChangeAdTypeHandler(event.target.value)}
+                            selected={values.type}
+                            required
+                        >
+                            <RadioGroup.Item value={buy_sell.BUY} label={localize('Buy {{currency}}', { currency })} />
+                            <RadioGroup.Item
+                                value={buy_sell.SELL}
+                                label={localize('Sell {{currency}}', { currency })}
+                            />
+                        </RadioGroup>
+                    )}
+                </Field>
+            )}
+            <div className='ad-type-section__container'>
+                <Field name='offer_amount'>
+                    {({ field }) => (
+                        <Input
+                            {...field}
+                            data-testid='offer_amount'
+                            data-lpignore='true'
+                            type='text'
+                            error={touched.offer_amount && errors.offer_amount}
+                            label={localize('Total amount')}
+                            className='ad-type-section__field'
+                            trailing_icon={
+                                <Text
+                                    color={is_desktop ? 'less-prominent' : 'prominent'}
+                                    size={is_desktop ? 'xxs' : 's'}
```

Comment:
can we create two variables for these two checks? cause they appear another 4 times in the file

Reply:
i think better to have a reusable component for this

Reply:
actually there is a utils function for this called `getTextSize` and `getIconSize` 😅 we can use that

Reply:
i mean not only for color and size but the entire component that we pass to `trailing_icon`

Reply:
ah yeah, we can do that and have a reusable component, you could move it up outside of the component, but in the same file